docs/modules/ROOT/pages/node_types.adoc

```diff
@@ -0,0 +1,235 @@
+* Assignment nodes: Any node that assigns a value. `lvasgn`, `ivasgn`, `cvasgn`, `gvasgn`, `send` (e.g. `self.foo = 5`), or `csend`.
+* Block statements: These can be essentially any node except those that must be nested (like `args` or `mlhs`). Typically they come as the children of a node that can contain arbitrary code, like `def` or `class`. If there are no statements, this will be a single `nil` node.
+
+== Location Information
+
+There are different parts of the source map you can get from the node, by calling `.loc` on it. Every node has an `expression` value which
+usually represents the entire node, but others have additional fields.
+
+The following fields are given when relevant to nodes in the source code:
+
+[cols="m,a"]
+|=========
+|Field|Description
+|assoc|The fat-arrow: `=>` used in `resbody` (NOT in hash `pair` s)
+|begin|* Start of parentheses or square brackets `(` or `[`
+* Start of `do..end` blocks (containing the `do` keyword)
+* `then` keyword
+* `begin` keyword
+* The first symbol in literals (like `"` for strings or `:` for symbols)
+|colon|The `:` symbol, used as part of a ternary (`:`)
+|dot|The `.` or `&.` operator used for `send` and `csend` nodes.
+|double_colon|The `::` operator
+|else|The `else` or `elsif` keyword
+|end|* End of parentheses or square brackets `)` or `]`
+* The `end` keyword
+|heredoc_body|The body of a string in heredoc format.
+|heredoc_end|The end of the heredoc statement.
+|in|The `in` keyword (as in `for..in`)
+|keyword|Any text-based keyword, like `begin` or `and`, with the exception of other keywords handled by other fields. This will include the full string containing the keyword.
+|name|Used when *defining* something (like `const` or `arg`).
+|operator|Any symbol representing an operator, like `*` (splat) or `\|\|=`.
+|question|The `?` symbol used as part of a ternary (`if`)
+|selector|Used for method invocation (`send` / `csend`) or operators that are actually methods (like `+`)
+
+|=========
+
+== Node Types
+
+[cols="m,a,a,m,m"]
+|=============================================
+|Type|Description|Children|Example|Node Class
+
+|alias|Method alias|Two children - both are `sym`, `dsym` or `gvar` nodes.|alias :foo :bar|https://rubydoc.info/github/rubocop-hq/rubocop-ast/RuboCop/AST/AliasNode[AliasNode]
+
+|and|And operator|Two children are both expression nodes representing the operands.|a and b|https://rubydoc.info/github/rubocop-hq/rubocop-ast/RuboCop/AST/AndNode[AndNode]
+
+|and_asgn|And-assignment (AND the receiver with the argument and assign it back to receiver).|First child must be an assignment node, second child is the expression node.|a &&= b|N/A
+
+|arg|Required positional argument. Must come inside an `args`.|1 child - a symbol, representing the argument name.|def foo(bar)|N/A
```

Comment:
I did my best for this with the new commit. It's not perfect because there are a few cases where there are variable arguments, but it's better :) Might make sense for me to just have a column with # of children as an int (or "variable") and then another one with description of children?